src/Assignments/ProblemIframe.tsx

```diff
@@ -138,41 +170,30 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         }
     }
 
-    function insertListener(problemForm: HTMLFormElement) {
-        const submitHandler = (clickedButton: HTMLButtonElement) => {
-            if (_.isNil(problemForm)) {
-                console.error('Hijacker: Could not find the form when submitting the form');
-                setError('An error occurred');
-                return;
-            }
-            if (_.isNil(clickedButton)) {
-                setError('Hijacker: An error occurred');
-                console.error('Could not find the button that submitted the form');
-                return;
-            }
-            prepareAndSubmit(problemForm, clickedButton);
-        };
-        const debouncedSubmitHandler = _.debounce(submitHandler, 4000, { leading: true, trailing: false });
+    function insertListeners(problemForm: HTMLFormElement) {
+        const throttledSubmitHandler = _.throttle(prepareAndSubmit, 2000, { leading: true, trailing: true });
+        const debouncedSubmitHandler = _.debounce(prepareAndSubmit, 2000, { leading: false, trailing: true });
 
+        // submission of problems will trigger updateSubmitActive @onLoad
+        // because re-submission of identical answers is blocked, we expect srcdoc to change
+        // even if our assumption fails, 'actual' submission is throttled
         problemForm.addEventListener('submit', (event: { preventDefault: () => void; }) => {
             event.preventDefault();
             const clickedButton = problemForm.querySelector('.btn-clicked') as HTMLButtonElement;
-            const shouldSubmit = (clickedButton.name === 'submitAnswers');
-            if (shouldSubmit) {
-                debouncedSubmitHandler(clickedButton);
+            if (clickedButton.name === 'submitAnswers') {
+                // submission is blocked until contents change - throttle will suffice
+                throttledSubmitHandler(problemForm, clickedButton);
```

Comment:
Why do we need this throttled, with these changes the user can't spam the submit button

Reply:
let's get rid of if else